internal/index/manager/manager.go

```diff
@@ -821,6 +829,27 @@ func (mgr *Manager) getIndexesCopy(start int) ([]*index.Reader, indexReleaser) {
 	return indexes, mgr.lock(indexes)
 }
 
+func (mgr *Manager) SetClientConfig(config ClientConfig) ClientConfig {
+	mgr.clientConfig = config
```

Comment:
We serialize changes and access from external code to the `Manager` instance through the `mgr.jobs` channel like you did below. Please use a job to retrieve the config too!